src/mlpack/methods/ann/activation_functions/identity_function.hpp

```diff
@@ -44,8 +44,9 @@ class IdentityFunction
    * @param x Input data.
    * @param y The resulting output activation.
    */
-  template<typename InputVecType, typename OutputVecType>
-  static void Fn(const InputVecType& x, OutputVecType& y)
+  template<typename VecType>
```

Comment:
It could be important to allow different input and output types here; for instance, if the input is `x.col(i)`, then `InputVecType` will be `arma::subview<eT>`, and if the output is some vector `y`, then the `OutputVecType` will be `arma::Col<eT>` or something like that.  So allowing parameters for both is some nice flexibility.

Reply:
Thanks, I get the reason why now. However, there is something I do not understand yet. If I am going to use `IsVector` SFINAE here, then it will only allow the same type for input and output right? or `IsVector` can accept `arma::subview`?